client/src/components/infoDrawer/infoDrawer.js

```diff
@@ -106,15 +122,101 @@ const singleValueCategoriesPlaceholder = Array.from(Array(NUM_CATEGORIES)).map(
   }
 );
 
+const renderDOILink = (type, doi, skeleton) => {
+  return (
+    doi && (
+      <>
+        <H3 className={skeleton ? Classes.SKELETON : null}>{type}</H3>
+        <p className={skeleton ? Classes.SKELETON : null}>
+          <a href={doi} target="_blank" rel="noopener noreferrer">
+            {doi}
+          </a>
+        </p>
+      </>
+    )
+  );
+};
+
+const affiliations = [];
+const renderContributors = (contributor) => {
```

Comment:
discussion:

For separation of concerns, maybe we can split the things this function does into multiple pieces, so it's easier to understand what it does and avoid side effect such as updating `affiliations` outside of the function's scope?

In theory we're doubling the runtime from N to 2N, but given the collection size is < 1000, readability + bug resistance likely outweighs the performance hit 😄 

E.g.,

```ts
const buildAffiliations = (contributors) => {
  const affiliations = contributors.map(
    (contributor) => contributor.institution
  );

  return Array.from(new Set(affiliations));
};

const renderContributors = (contributor, index) => {
  const { email, name } = contributor;

  return (
    <span key={name}>
      {name}
      {email && `(${email})`}
      <sup>{index + 1}</sup>
    </span>
  );
};

const InfoFormat = ({ ... }) => {
  if (dataPortalProps.corpora_schema_version === "1.0.0") {
    dataPortalProps = {};
  }

  const {
    title,
    publication_doi: doi,
    preprint_doi: preprintDOI,
    organism,
    contributors,
    project_links: projectLinks,
  } = dataPortalProps;

  const affiliations = buildAffiliations(contributors);

  ...
```